### PR TITLE
Components 2.0, now featuring atomized commits.

### PR DESCRIPTION
**Pull request by DominikPanic**
I'll need a lotta help with this.

### CODE REVIEW

File: code/datums/components/squeak.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		last_use = world.time
+		play_squeak()
+
+
+/datum/component/squeak/proc/on_equip(datum/source, mob/equipper, slot)
+	return
+
+
+/datum/component/squeak/proc/on_drop(datum/source, mob/user)
+	return
```
Comment (ninjanomnom) [inline code/datums/components/squeak.dm]: It's not like you don't have disposals, this is implementable

File: code/datums/datum.dm [hunk 2]
```diff
@@ -9,6 +10,7 @@
 	var/datum/weakref/weak_reference
 	var/hidden_from_codex = FALSE //set to TRUE if you want something to be hidden.
 
+
```
Comment (ninjanomnom) [inline code/datums/datum.dm]: You're missing all the code that cleans up components

File: code/game/atoms.dm [hunk 3]
```diff
@@ -691,12 +691,21 @@ Proc for attack log creation, because really why not
...
 	return INITIALIZE_HINT_NORMAL
 
+
 //called if Initialize returns INITIALIZE_HINT_LATELOAD
 /atom/proc/LateInitialize()
 	return
 
+
+// Put your AddComponent() calls here
+/atom/proc/ComponentInitialize()
```
Comment (ninjanomnom) [inline code/game/atoms.dm]: Don't port this part

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 4]
```diff
@@ -77,10 +81,15 @@
 	name = "clown shoes"
 	icon_state = "clown"
 	item_state = "clown_shoes"
-	slowdown = SHOES_SLOWDOWN+1
-	var/footstep = 1	//used for squeeks whilst walking
```
Comment (ninjanomnom) [inline code/modules/clothing/shoes/miscellaneous.dm]: I know you're replacing what this was for but was this not used at all? Where's the code removal of whatever is using this?

Comment (DominikPanic) [reply code/modules/clothing/shoes/miscellaneous.dm]: I couldn't find any reference to it anywhere.

File: code/datums/datum.dm (hunk 2 above)
Comment (DominikPanic) [reply code/datums/datum.dm]: Added

**Comment by ninjanomnom** [comment]
Seems probably fine but it needs to be tested

**Pull request merged**